sdk/azidentity/interactive_browser_credential.go

```diff
@@ -0,0 +1,117 @@
+// options: allow to configure the management of the requests sent to Azure Active Directory.
+func NewInteractiveBrowserCredential(tenantID string, clientID string, options *InteractiveBrowserCredentialOptions) (*InteractiveBrowserCredential, error) {
+	c, err := newAADIdentityClient(options.Options)
+	if err != nil {
+		return nil, err
+	}
+	var clientSecret *string
+	if options.ClientSecret != nil {
+		clientSecret = options.ClientSecret
+	}
+	return &InteractiveBrowserCredential{tenantID: tenantID, clientID: clientID, clientSecret: clientSecret, client: c}, nil
+}
+
+// GetToken obtains a token from Azure Active Directory using an interactive browser to authenticate.
+// ctx: Context used to control the request lifetime.
+// opts: TokenRequestOptions contains the list of scopes for which the token will have access.
+// Returns an AccessToken which can be used to authenticate service client calls.
+func (c *InteractiveBrowserCredential) GetToken(ctx context.Context, opts azcore.TokenRequestOptions) (*azcore.AccessToken, error) {
+	tk, err := c.client.authenticateInteractiveBrowser(ctx, c.tenantID, c.clientID, c.clientSecret, opts.Scopes)
+	if err != nil {
+		addGetTokenFailureLogs("Interactive Browser Credential", err, true)
+		return nil, err
+	}
+	logGetTokenSuccess(c, opts)
+	return tk, nil
+}
+
+// AuthenticationPolicy implements the azcore.Credential interface on InteractiveBrowserCredential and calls the Bearer Token policy
+// to get the bearer token.
+func (c *InteractiveBrowserCredential) AuthenticationPolicy(options azcore.AuthenticationPolicyOptions) azcore.Policy {
+	return newBearerTokenPolicy(c, options)
+}
+
+var _ azcore.TokenCredential = (*InteractiveBrowserCredential)(nil)
+
+var authCodeReceiver = func(tenantID string, clientID string, scopes []string) (*interactiveConfig, error) {
+	return interactiveBrowserLogin(tenantID, clientID, scopes)
+}
+
+// interactiveBrowserLogin opens an interactive browser with the specified tenant and client IDs provided then returns the authorization code
+// received or an error.
+func interactiveBrowserLogin(tenantID string, clientID string, scopes []string) (*interactiveConfig, error) {
+	const authURLFormat = "https://login.microsoftonline.com/%s/oauth2/v2.0/authorize?response_type=code&response_mode=query&client_id=%s&redirect_uri=%s&state=%s&scope=%s&prompt=select_account"
+	if tenantID == "" {
+		tenantID = "common"
+	}
+	if clientID == "" {
+		clientID = developerSignOnClientID
+	}
+	state := func() string {
+		rand.Seed(time.Now().Unix())
+		// generate a 20-char random alpha-numeric string
+		const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+		buff := make([]byte, 20)
+		for i := range buff {
+			buff[i] = charset[rand.Intn(len(charset))]
+		}
+		return string(buff)
+	}()
+	// start local redirect server so login can call us back
+	rs := newServer()
+	redirectURL := rs.Start(state)
```

Comment:
The redirectURL needs to be configurable to support the case that a user is using their own App Registration.

Reply:
Done, thanks for the info Scott!